test/e2e/framework/test_context.go

```diff
@@ -545,6 +550,31 @@ func AfterReadingAllFlags(t *TestContextType) {
 	}
 
 	if TestContext.ReportDir != "" {
+		// Create the directory before running the suite. If
+		// --report-dir is not unusable, we should report
+		// that as soon as possible. This will be done by each worker
+		// in parallel, so we will get "exists" error in most of them.
+		if err := os.MkdirAll(TestContext.ReportDir, 0777); err != nil && !os.IsExist(err) {
+			klog.Errorf("Create report dir: %v", err)
+			os.Exit(1)
+		}
```

Comment:
are we not already doing this here?

https://github.com/kubernetes/kubernetes/blob/ef5f1012f687cd633f29b5e9b2b1c7ded2145195/test/e2e/e2e.go#L226-L238

Reply:
Thanks for pointing that out. I searched the E2E *framework* for "os.Mkdir.*" and didn't find it. As `-report-dir` and report creation are features of the framework, that is where the directory should be created.

I also wonder whether this `reporters.NewJUnitReporter` actually does anything. Let me check...

Reply:
Your link is to old code, "master" only has the directory creation.

I removed it from test/e2e because this is now handled by the framework.

Reply:
oh, right, I didn't realize the search in github doesn't return the latest :)